manifests/server.pp

```diff
@@ -391,183 +391,123 @@
+  $puppet_basedir                                = $::puppet::server_puppet_basedir,
+  Optional[String] $puppetdb_host                = $::puppet::server_puppetdb_host,
+  $puppetdb_port                                 = $::puppet::server_puppetdb_port,
+  Boolean $puppetdb_swf                          = $::puppet::server_puppetdb_swf,
+  Enum['current', 'future'] $parser              = $::puppet::server_parser,
+  $environment_timeout                           = $::puppet::server_environment_timeout,
+  $jvm_java_bin                                  = $::puppet::server_jvm_java_bin,
+  $jvm_config                                    = $::puppet::server_jvm_config,
+  $jvm_min_heap_size                             = $::puppet::server_jvm_min_heap_size,
+  $jvm_max_heap_size                             = $::puppet::server_jvm_max_heap_size,
+  $jvm_extra_args                                = $::puppet::server_jvm_extra_args,
+  Stdlib::Absolutepath $jruby_gem_home           = $::puppet::server_jruby_gem_home,
+  Integer $max_active_instances                  = $::puppet::server_max_active_instances,
+  Integer $max_requests_per_instance             = $::puppet::server_max_requests_per_instance,
+  Boolean $use_legacy_auth_conf                  = $::puppet::server_use_legacy_auth_conf,
+  $check_for_updates                             = $::puppet::server_check_for_updates,
+  Boolean $environment_class_cache_enabled       = $::puppet::server_environment_class_cache_enabled,
 ) {
 
-  validate_bool($ca)
-  validate_bool($http)
-  validate_bool($passenger)
-  validate_bool($git_repo)
-  validate_bool($service_fallback)
-  validate_bool($server_foreman_facts)
-  validate_bool($strict_variables)
-  validate_bool($foreman)
-  validate_bool($puppetdb_swf)
-  validate_bool($default_manifest)
-  validate_bool($ssl_dir_manage)
-  validate_bool($ssl_key_manage)
-  validate_bool($passenger_pre_start)
-  validate_integer($passenger_min_instances)
-
-  validate_hash($additional_settings)
-
-  if $default_manifest {
-    validate_absolute_path($default_manifest_path)
-    validate_string($default_manifest_content)
-  }
-
-  validate_string($hiera_config)
-  validate_string($external_nodes)
-  if $ca_proxy {
-    validate_string($ca_proxy)
-  }
-  if $puppetdb_host {
-    validate_string($puppetdb_host)
-  }
-
-  if $http {
-    validate_array($http_allow)
-  }
-
   if ! is_bool($autosign) {
     validate_absolute_path($autosign)
     validate_string($autosign_mode)
     validate_array($autosign_entries)
   }
 
-  if $autosign_content {
-    validate_string($autosign_content)
-  }
-
-  validate_array($rack_arguments)
-
-  validate_re($implementation, '^(master|puppetserver)$')
-  validate_re($parser, '^(current|future)$')
-
   if $environment_timeout {
     validate_re($environment_timeout, '^(unlimited|0|[0-9]+[smh]{1})$')
   }
 
   if $implementation == 'puppetserver' {
     validate_re($jvm_min_heap_size, '^[0-9]+[kKmMgG]$')
```

Comment:
I think this is correct:
https://docs.puppet.com/puppet/4.6/lang_data_abstract.html#pattern